### PR TITLE
Version docs for v0.10.0

### LINKED ISSUE #882: Introduce versioning of documentation site to TCE
Some of what's involved for how Velero does it:
https://velero.io/docs/v1.6/release-instructions/#changelog-and-docs-pr
https://github.com/vmware-tanzu/velero/blob/main/Makefile#L351-L361
https://github.com/vmware-tanzu/velero/blob/main/hack/release-tools/gen-docs.sh#L48

### PR DESCRIPTION
**Pull request by kcoriordan**
## What this PR does / why we need it

This PR versions the doc for v0.10.0. Proposal is that the docs will now default to point to ` /v0.10.0`. We will continue to work in the` /latest` for the next release and at next release time we will create a  folder `<next-release>` based on contents of  `/latest` and change the docs to default to `<next-release>` . 

1. In `config.yaml `  added `v0.10.0 ` under `doc versions`  and `release_versions`,   and changed `doc_latest`  to `v0.10.0 `

1. Created a new folder  `/v0.10.0`
2. Moved/copied files as follows
        a. Copied all the files  from `/latest` including sub folders  into `/v0.10.0`
        b. Copied `/assets `into` /v0.10.0`
        c. Copied `/assets` into` /latest`
        d. Updated paths for includes in all files to reflect new folder structure
        e. Updated `check-mdlint.sh` to ignore the `/assets` folders in new location

3. .Created a `v0.10.0-toc.yml` version of toc and added a mapping in `toc-mapping.yml` .
4. In `v0.10.0/index.md` changed version value to `v0.10.0 `.

Preview here: https://deploy-preview-3212--quirky-franklin-8969be.netlify.app/docs/latest/cli-installation/

## Details for the Release Notes (PLEASE PROVIDE)

```release-note

```

## Which issue(s) this PR fixes

Fixes: #882

## Describe testing done for PR
Run hugo server and preview here: https://deploy-preview-3212--quirky-franklin-8969be.netlify.app/

## Special notes for your reviewer
Netlify preview link is: https://deploy-preview-3212--quirky-franklin-8969be.netlify.app/

### CODE REVIEW

**Comment by kcoriordan** [comment]
@jonasrosland

File: docs/site/content/docs/latest/assets/cli-install-mac.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ You can install Tanzu Community Edition using Homebrew **or** you can download t
 
     ```sh
     cd <DOWNLOAD-DIR>
-    tar xzvf tce-darwin-amd64-{{< release_latest >}}.tar.gz
+    tar xzvf ~/tce-darwin-amd64-{{< release_latest >}}.tar.gz
```
Comment (dvonthenen) [inline docs/site/content/docs/latest/assets/cli-install-mac.md:29]: The `~/` assumes you extracted everything in your homedir. I think the correct think would be extract in the directory you downloaded into.

**Comment by dvonthenen** [comment]
@kcoriordan  Based on the procedure you have written above, we could revisit/revive the automation to bump the versions and also do the steps above when the release is tagged.

If this is implemented, this does mean when the repo is tagged for the final GA, any changes for that `GA release (ie v0.10.0)` would need to be duplicated in the `latest` and `GA release` folders.

**Review by jonasrosland — DISMISSED** [review]
This looks good to me @kcoriordan !

**Comment by stmcginnis** [comment]
This works well, but I really do not like the duplication of all of our content. That is extra bloat, and it's duplication of content that's already in our branches (where someone would reasonably expect to look for versioned doc updates).

Have you explored something like what is discussed in option 2 here: https://answers.netlify.com/t/versioned-netlify-deploys/3050/2

That points to how it would be done in netlify here: https://answers.netlify.com/t/support-guide-can-i-deploy-multiple-repositories-in-a-single-site/179

Basically, we would just need a drop down that is aware of the different versions to be able to switch between them, but we can continue to support docs updates for past releases the same way we would handle code changes.

**Comment by kcoriordan** [comment]
using sub domains based on branches would remove the need for duplication which would be good
At the moment Netlify is setup to only publish the production branch
So I can see where to change this around in Netlify, 
but not sure how to enable the drop down, or the default branch/sub domain

@jonasrosland

**Comment by jonasrosland** [comment]
Versioning through Netlify works, you can see it live at https://v1-22.docs.kubernetes.io/docs/home/supported-doc-versions/ for instance.

If you want to version through release tags we can also do that through a deployment script we use for Harbor:
https://github.com/goharbor/website/blob/main/load-docs.sh
That negates the need for subdomains, and dropdown version filtering is easy.

**Comment by dvonthenen** [comment]
I notice there was a discussion in slack about this... let me know if there is any action on our side or if this is ready for review. Will look for @jonasrosland or @kcoriordan for guidance

**Comment by stmcginnis** [comment]
Oops, should have captured some of the notes from that discussion here. Latest status is we are going to look at adapting some of the scripting done in Harbor for use in TCE. That way we can keep versioned docs on the `release-0.foo.0` branches, and we don't need to duplicate and accumulate copies of the docs as we go.

@jonasrosland was going to work on updating some of the scripting and work with @kcoriordan to get an update here.

**Comment by joshrosso** [comment]
> #3377 now to replace this PR

@kcoriordan can this PR be closed then?

If so, feel free to close this.

**Comment by kcoriordan** [comment]
closing
work to version docs in these PRs:
https://github.com/vmware-tanzu/community-edition/pull/3490
https://github.com/vmware-tanzu/community-edition/pull/3558
